smart-contracts/src/main/scala/io/casperlabs/smartcontracts/cltype/ProtoMappings.scala

```
@@ -0,0 +1,149 @@
+
+    case StoredValue.Contract(contract) =>
+      Some(state.Value(state.Value.Value.Contract(toProto(contract))))
+  }
+
+  def toProto(c: Contract): state.Contract = state.Contract(
+    ByteString.copyFrom(c.bytes.toArray),
+    toProto(c.namedKeys),
+    Some(toProto(c.protocolVersion))
+  )
+
+  def toProto(a: Account): state.Account = state.Account(
+    ByteString.copyFrom(a.publicKey.bytes.toArray),
+    toProto(Key.URef(a.purseId)).value.uref,
+    toProto(a.namedKeys),
+    a.associatedKeys.toSeq.map {
+      case (k, w) => state.Account.AssociatedKey(ByteString.copyFrom(k.bytes.toArray), w.toInt)
+    },
+    Some(
+      state.Account
+        .ActionThresholds(
+          a.actionThresholds.deployment.toInt,
+          a.actionThresholds.keyManagement.toInt
+        )
+    )
+  )
+
+  def toProto(namedKeys: Map[String, Key]): Seq[state.NamedKey] = namedKeys.toSeq.map {
+    case (n, k) => state.NamedKey(n, Some(toProto(k)))
+  }
+
+  def toProto(k: Key): state.Key = k match {
+    case Key.Account(address) =>
+      state.Key(
+        state.Key.Value.Address(state.Key.Address(ByteString.copyFrom(address.bytes.toArray)))
+      )
+
+    case Key.Hash(address) =>
+      state.Key(
+        state.Key.Value.Hash(state.Key.Hash(ByteString.copyFrom(address.bytes.toArray)))
+      )
+
+    case Key.URef(uref) =>
+      state.Key(
+        state.Key.Value.Uref(state.Key.URef(ByteString.copyFrom(uref.address.bytes.toArray)))
+      )
+
+    case Key.Local(address) =>
+      state.Key(
+        state.Key.Value.Local(state.Key.Local(ByteString.copyFrom(address.bytes.toArray)))
+      )
+  }
+
+  def toProto(version: SemVer): state.ProtocolVersion = state.ProtocolVersion(
+    version.major,
+    version.minor,
+    version.patch
+  )
+
+  def toProto(v: CLValue): Option[state.Value] = v.clType match {
+    case CLType.I32 | CLType.U32 =>
+      FromBytes.deserialize[Int](v.value.toArray).toOption.map { i =>
```

Are all these `.toOption` values masking errors? But those errors are not because `state.Value` can't represent the result, but for some other reason? I think as long as the only way to read data is through `state.Value` it would be better to return whatever's happening rather than conflate the issue.